Makefile

```diff
@@ -103,13 +115,12 @@ install-doc:
 	install -m644 dunstrc ${DESTDIR}${PREFIX}/share/dunst
 
 install-service: service
-	mkdir -p ${DESTDIR}${PREFIX}/share/dbus-1/services/
-	install -m644 org.knopwob.dunst.service ${DESTDIR}${PREFIX}/share/dbus-1/services
-	install -Dm644 dunst.systemd.service ${DESTDIR}${PREFIX}/lib/systemd/user/dunst.service
+	install -Dm644 org.knopwob.dunst.service ${DESTDIR}${SERVICEDIR_DBUS}/org.knopwob.dunst.service
```

Comment:
It's a fatal error if one of these install calls fail, the error condition should be handled both here and in `install-systemd` and also in `uninstall-{dbus,systemd}` since it currently stops in the middle of the installation (or uninstallation).

PS: Why is the install target called `install-service` but uninstall `uninstall-dbus`? weird.

Reply:
Naming issue: true, ack.

Permission issue: what!? It's common to start and not check. However it's handled as fatal (like every command in a Makefile).

It's not the makefile's job to check for succession before installation. Also it's (that'll sound hard) not our job to make error messages user friendly. Anyone who's capable to execute make should understand what a permission denied error is.

Reply:
Consider this scenario:
```
$ PREFIX=install/ SERVICEDIR_DBUS=install/dbus make install
install -Dm755 dunst install//bin/dunst
install -Dm644 docs/dunst.1 install//share/man/man1/dunst.1
install -Dm644 dunstrc install//share/dunst/dunstrc
install -Dm644 org.knopwob.dunst.service install/dbus/org.knopwob.dunst.service
install -Dm644 dunst.systemd.service /usr/lib/systemd/user/dunst.service
install: cannot remove '/usr/lib/systemd/user/dunst.service': Permission denied
Makefile:136: recipe for target 'install-service-systemd' failed
make: *** [install-service-systemd] Error 1
```
The fact that this means that the installation was successful but couldn't install a very optional part is weird right off the bat.

```
$ find install/
install/
install/bin
install/bin/dunst
install/share
install/share/man
install/share/man/man1
install/share/man/man1/dunst.1
install/share/dunst
install/share/dunst/dunstrc
install/dbus
install/dbus/org.knopwob.dunst.service
```
All normal here

```
$ PREFIX=install/ SERVICEDIR_DBUS=install/dbus make uninstall
rm -f install/dbus/org.knopwob.dunst.service
rm -f /usr/lib/systemd/user/dunst.service
rm: cannot remove '/usr/lib/systemd/user/dunst.service': Permission denied
Makefile:150: recipe for target 'uninstall-systemd' failed
make: *** [uninstall-systemd] Error 1
```

Looks like something didn't run here and indeed uninstallation finished midway through
```
$ find install/
install/
install/bin
install/bin/dunst
install/share
install/share/man
install/share/man/man1
install/share/man/man1/dunst.1
install/share/dunst
install/share/dunst/dunstrc
install/dbus
```

The only reason the `install` command worked correctly is because `install-systemd` happened to be the last target to be run it'd have exited and might've skipped some steps otherwise.

It's common not to check permissions when we would fail anyway if we can't install that specific part but it's not what's happening here, this is completely optional and it's making it look like something fatal happened.

Reply:
Sorry, but I have to fully disagree with you. Let me tell you, why:

Let's make an axiom test first.

- (1) In an unparametrized make call: systemd available <=> systemd service file installed,
- (2) If you have systemd installed, you want to get valuable information from `systemctl status --user dunst.service`

Please validate all your statements of your previous comment against those two axioms.

Reply:
I was going to argue the other way but then I realised it's the exact same situation with the dbus service which is pretty crucial. So I give it a pass here.

What I still don't like though is the uninstallation failing mid-way, the same way we use -f to ignore errors on non-existent files we should check and only issue a big warning if there is a permission issue with one of the files not stop completely and leave junk files behind.

Reply:
> What I still don't like though is the uninstallation failing mid-way,

Err... Failing the installation midway implies, that the installation did not succeed. So, you can't uninstall dunst successfully if your installation failed.

Packing all uninstall targets into a single one and in combination with some shell foo, this can get solved. This ignores all make principles, but it's the only possible solution. I've pushed a commit to this branch now. Let's discuss it.